src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -463,4 +506,50 @@ static EpInfo buildEpInfoFromStats(long rowCount, Map<String, RowBufferStats> co
     }
     return epInfo;
   }
+
+  /** Row buffer factory. */
+  static <T> AbstractRowBuffer<T> createRowBuffer(BufferConfig bufferConfig) {
+    // TODO: The circular dependency SnowflakeStreamingIngestChannelInternal <-> RowBuffer
+    // (SNOW-657667)
+    // can be probably reconsidered
+    switch (bufferConfig.bdecVersion) {
+      case ONE:
+        //noinspection unchecked
+        return (AbstractRowBuffer<T>) new ArrowRowBuffer(bufferConfig);
+      case THREE:
+        //noinspection unchecked
+        return (AbstractRowBuffer<T>) new ParquetRowBuffer(bufferConfig);
+      default:
+        throw new SFException(
+            ErrorCode.INTERNAL_ERROR,
+            "Unsupported BDEC format version: " + bufferConfig.bdecVersion);
+    }
+  }
+
+  static class BufferConfig {
```

Comment:
comment?

Reply:
And could we add getter function for all the fields?

Reply:
I can do this but for such internal classes, it feels like just cluttering the code.